remove lodash

aliceriot opened the pull request:
this removes lodash 

fixes #3 

unfortunately though, it looks like lodash is a dependency of a lot of our dependencies, so it might still get pulled in when installing.

noisecapella commented on src/util.js:
```diff
@@ -18,5 +17,5 @@ export const withUsername = (type: ActionType, payloadFunc: Function = R.nthArg(
 
 export const updateStateByUsername = (state: Object, username: string, update: Object) => (
-  _.merge({}, state, { [username]: update })
+  Object.assign({}, state, { [username]: update })
```
Can we use the spread syntax here instead?

aliceriot commented:
I changed that and added a few tests as well

aliceriot commented:
I'll go ahead and merge if you think that looks good!

noisecapella commented:
:+1:

aliceriot commented:
I just realized I forgot one thing, I'll ping you when I address it

aliceriot commented:
I forgot we were using `_.merge` so that updates would be merged into the existing namespaced (per-username) state. So I added a test to make sure that's happening, and added the `lodash.merge` package so we can get just that method.

aliceriot commented:
@noisecapella the changes are pretty trivial but if you wanna take a look before I merge that would be :100:
